pkg/sdk/alerts.go

```diff
@@ -0,0 +1,382 @@
+	ShowByID(ctx context.Context, id SchemaObjectIdentifier) (*Alert, error)
+	// Describe returns the details of an alert.
+	Describe(ctx context.Context, id SchemaObjectIdentifier) (*AlertDetails, error)
+}
+
+// alerts implements Alerts
+type alerts struct {
+	client *Client
+}
+
+type CreateAlertOptions struct {
+	create      bool                   `ddl:"static" sql:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace   *bool                  `ddl:"keyword" sql:"OR REPLACE"`
+	alert       bool                   `ddl:"static" sql:"ALERT"` //lint:ignore U1000 This is used in the ddl tag
+	IfNotExists *bool                  `ddl:"keyword" sql:"IF NOT EXISTS"`
+	name        SchemaObjectIdentifier `ddl:"identifier"`
+
+	// required
+	warehouse AccountObjectIdentifier `ddl:"identifier,equals" sql:"WAREHOUSE"`
+	schedule  string                  `ddl:"parameter,single_quotes" sql:"SCHEDULE"`
+
+	// optional
+	Comment *string `ddl:"parameter,single_quotes" sql:"COMMENT"`
+
+	// required
+	condition AlertCondition `ddl:"condition,parentheses"   sql:"IF"`
+	action    string         `ddl:"parameter,no_equals" sql:"THEN"`
+}
+
+type AlertCondition struct {
+	Condition string `ddl:"condition,parentheses" sql:"EXISTS"`
+}
+
+func (opts *CreateAlertOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return errors.New("invalid object identifier")
+	}
+
+	return nil
+}
+
+func (v *alerts) Create(ctx context.Context, id SchemaObjectIdentifier, warehouse AccountObjectIdentifier, schedule string, condition string, action string, opts *CreateAlertOptions) error {
+	if opts == nil {
+		opts = &CreateAlertOptions{}
+	}
+	opts.name = id
+	opts.warehouse = warehouse
+	opts.schedule = schedule
+	opts.name = id
+	opts.condition = AlertCondition{condition}
+	opts.action = action
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type AlertOperation string
+
+var (
+	// Resume makes a suspended alert active.
+	Resume AlertOperation = "RESUME"
+	// Suspend puts the alert into a “Suspended” state.
+	Suspend AlertOperation = "SUSPEND"
+)
+
+type AlertState string
+
+var (
+	Started   AlertState = "started"
+	Suspended AlertState = "suspended"
+)
+
+type AlterAlertOptions struct {
+	alter    bool                   `ddl:"static" sql:"ALTER"` //lint:ignore U1000 This is used in the ddl tag
+	alert    bool                   `ddl:"static" sql:"ALERT"` //lint:ignore U1000 This is used in the ddl tag
+	IfExists *bool                  `ddl:"keyword" sql:"IF EXISTS"`
+	name     SchemaObjectIdentifier `ddl:"identifier"`
+
+	// One of
+	Operation       *AlertOperation `ddl:"keyword"`
+	Set             *AlertSet       `ddl:"keyword" sql:"SET"`
+	Unset           *AlertUnset     `ddl:"keyword" sql:"UNSET"`
+	ModifyCondition *string         `ddl:"condition,parentheses" sql:"MODIFY CONDITION EXISTS"`
```

Comment:
again, this could be done using a list type. or adding parentheses modifier to parameter type. so then it would be like `ddl:"parameter,parantheses,no_equals" sql:"MODIFY CONDITION EXISTS"`

Reply:
fixed using `list` type, but that means instead of:
```
alterOptions.ModifyCondition = &condition
```
I need to do 
```
alterOptions.ModifyCondition = &[]string{condition}
```
In this case it's not a big problem, but I was wondering if you considered using separate types for constructing sql statements(structs with `ddl` tags) and for arguments inside client's interface.

Structs with ddl struct tags need to have a certain structure so that we can turn them into SQL statements, but since the same type is used in client's interface, we allow users of that interface to make a mistake, e.g. passing multiple conditions to `ModifyCondition`

I think this might prove useful if we were to convert sdk package into a stand-alone library in the future.
I'm not saying we should make this change now, but I am curious what are your thoughts on it.